File: Lab2/javaServer/src/main/java/com/JabaJabila/javaServer/dto/CatViewModel.java
package com.JabaJabila.javaServer.dto;
import com.JabaJabila.javaServer.entities.Cat;

import java.util.Date;

public class CatViewModel {
    private Long catId;
    private String name;
    private String breed;
    private Date birthdate;
    private Long ownerId;

    public Long getCatId() {
        return catId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBreed() {
        return breed;
    }

    public void setBreed(String breed) {
        this.breed = breed;
    }

    public Date getBirthdate() {
        return birthdate;
    }

    public void setBirthdate(Date birthdate) {
        this.birthdate = birthdate;
    }

    public Long getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(Long ownerId) {
        this.ownerId = ownerId;
    }

    public static CatViewModel CreateFromEntity(Cat cat) {
        CatViewModel result = new CatViewModel();
        result.catId = cat.getCatId();
        result.name = cat.getName();
        result.breed = cat.getBreed();
        result.birthdate = cat.getBirthdate();

        if (cat.getOwner() != null)
            result.ownerId = cat.getOwner().getOwnerId();
        else
            result.ownerId = null;

        return result;
    }
}

File: Lab2/Codegen/MyHttpClient/GeneratedModels/CatViewModel.cs
namespace MyHttpClient.GeneratedModels;
public class CatViewModel
{
    public CatViewModel(long catId, string name, string breed, JavaDateFormat birthdate, long ownerId)
    {
        CatId = catId;
        Name = name;
        Breed = breed;
        Birthdate = birthdate;
        OwnerId = ownerId;
    }

    public long CatId { get; set; }

    public string Name { get; set; }

    public string Breed { get; set; }

    public JavaDateFormat Birthdate { get; set; }

    public long OwnerId { get; set; }
}